### PR TITLE
use crazy cross-version helper registration

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

File: addon/lib/helpers.js [hunk 1]
```diff
@@ -1,102 +1,59 @@
 import Ember from 'ember';
-import {singularize, pluralize} from "./system/string";
+import {
+  singularize,
+  pluralize
+} from "./system/string";
 
-if (Ember.HTMLBars) {
```
Comment (rwjblue) [inline addon/lib/helpers.js]: Still need this conditional.  The registerHelper imported below is ONLY for if `Ember.HTMLBars` exists.

Comment (stefanpenner) [inline addon/lib/helpers.js]: i should be able to do this within registry-helper c/d ?

**Comment by bmac** [comment]
Should these helpers still be conditionally registering on Ember.Handlebars.helper for old versions of Ember?

**Comment by stefanpenner** [comment]
Ya. I'll fix that shortly. Also going I try @katiegenglar s new addon. See if we can test multiple versions

File: addon/lib/helpers.js (hunk 1 above)
Comment (stefanpenner) [inline addon/lib/helpers.js]: i think I have handled the cases correctly in the helper now.

**Comment by stefanpenner** [comment]
when ember-try is ready, we can use it to better cover these cases https://github.com/stefanpenner/ember-inflector/issues/72

**Comment by stefanpenner** [comment]
@rwjblue @bmac r?

**Comment by bmac** [comment]
lgtm :+1:

File: addon/lib/utils/register-helper.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import Ember from 'ember';
+
+function registerHelperIteration1(name, helperFunction) {
+  //earlier versions of ember with htmlbars used this
+  Ember.HTMLBars.helpers[name] = helperFunction;
```
Comment (sly7-7) [inline addon/lib/utils/register-helper.js]: @stefanpenner So it looks like https://github.com/emberjs/list-view/blob/master/addon/helper.js#L8 is missing this one. Should we add it to list-view as well ?

Comment (stefanpenner) [inline addon/lib/utils/register-helper.js]: yes that would be awesome. (maybe we should consider this some micro-lib we share)

Comment (sly7-7) [inline addon/lib/utils/register-helper.js]: I think I can do that, at least try. Though I have no idea how do you want this micro-lib (an ember-cli addon itself ?)

Comment (stefanpenner) [inline addon/lib/utils/register-helper.js]: nah, just an npm module i think.

Comment (sly7-7) [inline addon/lib/utils/register-helper.js]: Well I firstly tried to basically use this code in list-view, but it did not work, I encountered `Assertion Failed: makeBoundHelper generated helpers do not support use with blocks`. So I guess we can't just share it :(. Still I can at least add the missing `Ember.HTMLBars.helpers`case.

**Pull request merged**